tenantable_belongs_to method

ptarud opened the pull request:
Add a tenantable_belongs_to class method that automatically validates that the parent association must has the same tenant than the record is defining the tenantable_belongs_to association.

ptarud commented:
@excid3 Hi, I hope you are very well. Can you review this PR please ?

mrosso10 commented:
Hi @ptarud, what would be the difference with the already existing validation?

https://github.com/ErwinM/acts_as_tenant/blob/7e3bd8a534f309061a9c34b4ed3eb3576e2e065d/lib/acts_as_tenant/model_extensions.rb#L44

mrosso10 commented:
Nevermind my previous comment, I saw the difference, and actually found this useful for admin panels where there is no current_tenant.

Thanks!

ptarud commented:
When there is tenant setted and you change the tenant programmatically doing ActsAsTenant.current_tenant = other_tenant
and then for example you have a Parent class instanciated and then you created a Child class (that belongs_to :parent) you can assign the parent into that other child and the tenants will does not match. 

Take a look of this peace of code:

```ruby
ActsAsTenant.current_tenant = Tenant.first
t1_company = FactoryBot.create(:company)
ActsAsTenant.current_tenant = Tenant.second
t2_branch = FactoryBot.create(:branch, company: t1_company)

t1_company.inspect
=> "#<Company id: 1, tenant_id: 1, fantasy_name: \"Fantasy Name 1\", deleted: false, created_at: \"2024-09-06 15:58:45.439919000 +0000\", updated_at: \"2024-09-06 15:58:45.439919000 +0000\">"

t2_branch.inspect
=> "#<Branch id: 1, tenant_id: 2, company_id: 1, name: \"Branch 1\", time_zone: \"Etc/UTC\", deleted: false, created_at: \"2024-09-06 16:00:01.057926000 +0000\", updated_at: \"2024-09-06 16:00:01.057926000 +0000\">"
```

I think is useful having a functionality for checking this behaviour.

mrosso10 commented:
Hi @ptarud, what do you think about optionally auto-assigning the tenant id from the associated record?

Just implemented it here, feel free to cherry pick:
https://github.com/mrosso10/acts_as_tenant/commit/5e21cda240ef98ce10763d25b37913bef79b29fb
